src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ARRAY_MULTIPLIER = 2;
+    private Object[] elements;
```

Comment:
```suggestion
    private T[] elements;
```

Reply:
it is not necessary to create an Object immediately. And fix the relevant errors related to this